providers/gce/gce_instances.go

```diff
@@ -92,6 +93,58 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 	return refs
 }
 
+// cmpAddressFamily compares IPv6 and IPv4 address families so that IPv6 is smaller than IPv4.
```

Comment:
This is not a sort problem, is a filter problem, we need to filter ipv4 addresses and IPv6 addresses and add ones or.the others first.depending on the stack type ... The order of the addresses in each family has to be preserved

Reply:
Internal addresses will be before external ones (this is how it is implemented right now). Is it not enough?

Reply:
I changed to stable sorting.

Reply:
I changed the approach so now it only sorts by ip family.

Reply:
There is no sorting any more.